### ISSUE TITLE
Unable to open any models with new 2.0 version

### ISSUE THREAD

**Issue by Exitaph** [Gravemind2401/Reclaimer#76]
I am unable to open any models from any map from any Halo game with either of the currently available 2.0 versions of Reclaimer. Upon double clicking an asset, Reclaimer freezes up for a few seconds, then crashes. I have installed the new .NET runtime. The old version 1.7.840 still works just fine. 

I also tried uninstalling all versions with Revo Uninstaller, then reinstalling 2.0.1509. Same behavior. Then updating to 2.0.1536, same behavior.

**Comment by Gravemind2401** [comment]
After the app crashes, if you reopen it and go to `View > AppData Directory` from the menu there should be a `crash.txt` file where the data modified is the time it crashed. If you attach a copy of that file it may help identify the issue.

Also, if you go to `View > App Directory` it will take you to the app install folder. From there, check that the following files all exist and have the correct version (right click the file, select properties, then look at the Details tab in the properties window):
- Reclaimer.exe
- Reclaimer.dll
- Reclaimer.Blam.dll
- Reclaimer.Core.dll

If any of them do not same version number as the installer you used then the installer may have had trouble replacing existing files. You can try deleting that whole folder then run the installer again and use the Repair option.

**Comment by Exitaph** [comment]
Unfortunately no crash.txt is being created. The only items in AppData are "settings.json" and a folder named "Plugins" which is empty. I should have mentioned that the program hard crashes with no error message or warning.

- Reclaimer.exe - 2.0.1536.0
- Reclaimer.dll - 2.0.1536.0
- Reclaimer.Blam.dll - 2.0.1536.0
- Reclaimer.Core.dll - 2.0.1536.0

Looks like they are all the correct version.

**Comment by Gravemind2401** [comment]
If there is no crash file you may be able to get the crash details from the event viewer. You can follow the steps in this video to find the crash in event viewer:
https://www.youtube.com/watch?v=TK768PB0BGc

**Comment by Exitaph** [comment]
Thanks for the tip, that did it. Here is the event error information that it gives. Appears to be and issue with nvwgf2umx.dll. After a quick google search this appears to be a common problem with games and/or 3d applications. 

Faulting application name: Reclaimer.exe, version: 2.0.1536.0, time stamp: 0x66470000
Faulting module name: nvwgf2umx.dll, version: 32.0.15.6614, time stamp: 0x672bc78e
Exception code: 0xc00000fd
Fault offset: 0x0000000000f20887
Faulting process id: 0xc78
Faulting application start time: 0x01db459821bba193
Faulting application path: C:\ProgramData\Gravemind2401\Reclaimer\Reclaimer.exe
Faulting module path: C:\WINDOWS\System32\DriverStore\FileRepository\nv_dispsi.inf_amd64_3d88c2eb4775cc07\nvwgf2umx.dll
Report Id: 1679f00a-d6cb-4a27-8d75-29583c8045f8
Faulting package full name: 
Faulting package-relative application ID:

**Comment by Gravemind2401** [comment]
The crash is happening in the display drivers, which unfortunately there's not much I can do to diagnose or resolve. It could be related to the studio drivers there could just be some issue with the specific version of the drivers you have.

You could try updating the drivers if there are any updates available, or if you feel like getting a headache you could try switching from studio drivers to game ready drivers and see if that's any different.

The old version of Reclaimer likely doesn't have the issue because it uses WPF's basic built-in 3D rendering. The new version uses a package called HelixToolkit instead, which uses DirectX based rendering.

**Comment by Exitaph** [comment]
Yeah I figured that was the case. But now I have a path to troubleshoot. I tried doing a clean install of the drivers but have not tried going back to game ready drivers. That or it might just be time for a fresh windows install. Thank you for your help though and thank you for such a great piece of software. Luckily the old version still works and I was also able to get the new version working on another rig. I really like the new export/import format/workflow.
